app/src/main/java/org/efidroid/efidroidmanager/LokiTool.java

```diff
@@ -0,0 +1,73 @@
+package org.efidroid.efidroidmanager;
+
+import android.content.Context;
+import android.os.Build;
+
+import org.apache.commons.io.FileUtils;
+import org.efidroid.efidroidmanager.services.IntentServiceEx;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class LokiTool {
+    public enum PartitionLabel {
+        BOOT("boot"),
+        RECOVERY("recovery");
+
+        private final String mLabel;
+
+        PartitionLabel(String s) {
+            mLabel = s;
+        }
+
+        public String getLabel() {
+            return mLabel;
+        }
+    }
+
+    private LokiTool() {
+    }
+
+    public static void init(Context context) {
+        ArrayList<String> abis = new ArrayList<>();
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+            abis.addAll(Arrays.asList(Build.SUPPORTED_ABIS));
+        } else {
+            abis.add(Build.CPU_ABI);
+            abis.add(Build.CPU_ABI2);
+        }
+        // do not initialize on non-arm cpu
+        if (!abis.contains("armeabi-v7a")) return;
```

Comment:
This tool is usable only for arm-based phones. 
See https://github.com/efidroid/modules_loki/blob/master/loki_patch.c#L30